atlasdb-remoting-api/src/main/java/com/palantir/atlasdb/config/AuxiliaryRemotingParameters.java

```diff
@@ -48,7 +48,11 @@ default boolean shouldSupportBlockingOperations() {
         return true;
     }
 
+    /**
+     * @deprecated Conjure is now always used, and supplying alternative configs is not possible.
```

Comment:
I think we should go through and purge the config stuff properly. Deprecations don't really do anything if we never remove them.

I also don't think we want to deprecate this *anyway*. Since if we do need to open up other configurations i.e. if we want to add some configurability for our own testing purposes, this is the right place to do it.

Starting from the config `RemotingClientConfig`, we can delete the two properties, and if we're super concerned we can do what we did in `AtlasDbConfig`, where as a class level annotation, we do `@JsonIgnoreProperties({"maximumConjureRemotingProbability", "enableLegacyClientFallback"}` and then we delete those properties.

We should probably add a `RemotingClientConfigs.DEFAULT` configuration, and then have the old thing `RemotingClientConfigs.ALWAYS_USE_CONJURE` refer to that, and change all of our references to that.

Then we can propagate the deletion of the java properties everywhere (i.e. and continue to get rid of `VersionSelectingClients`.